spec/migration/parity_check/dynamic_request_content_spec.rb

```diff
@@ -69,6 +69,41 @@
       it { is_expected.to eq(partnership.api_id) }
     end
 
+    context "when fetching `teacher_api_id`", :with_metadata do
+      let(:identifier) { :teacher_api_id }
+      let(:lead_provider_delivery_partnership) { FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider:) }
+      let(:school_partnership) { FactoryBot.create(:school_partnership, active_lead_provider:) }
+      let!(:training_period) { FactoryBot.create(:training_period, :for_ect, :ongoing, school_partnership:) }
+      let(:teacher) { training_period.trainee.teacher }
+
+      before do
+        # Participants for different lead providers should not be used.
+        FactoryBot.create(:training_period, :for_ect, :ongoing)
+        FactoryBot.create(:training_period, :for_mentor, :ongoing)
+      end
+
+      it { is_expected.to eq(teacher.api_id) }
+    end
+
+    context "when fetching `from_teacher_api_id`", :with_metadata do
+      let(:identifier) { :from_teacher_api_id }
+      let(:lead_provider_delivery_partnership) { FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider:) }
+      let(:school_partnership) { FactoryBot.create(:school_partnership, active_lead_provider:) }
+      let(:training_period) { FactoryBot.create(:training_period, :for_ect, :ongoing, school_partnership:) }
+      let(:teacher) { training_period.trainee.teacher }
+      let(:from_teacher) { FactoryBot.create(:teacher) }
+
+      before do
+        FactoryBot.create(:teacher_id_change, teacher:, api_from_teacher_id: from_teacher.api_id)
+
+        # Participants for different lead providers should not be used.
+        FactoryBot.create(:training_period, :for_ect, :ongoing)
```

Comment:
Should these be other `teacher_id_change` records not associated to `from_teacher`?